agent/lib_monolog.c

```diff
@@ -295,17 +310,23 @@ static char* nr_monolog_get_context(const size_t argc,
  */
 static char* nr_monolog_build_message(const size_t argc,
                                       NR_EXECUTE_PROTO TSRMLS_DC) {
+#if !HAVE_CONTEXT_IN_MESSAGE
+  /* Make the compiler happy - argc is not used when $context is ignored */
+  (void)argc;
+#endif
```

Comment:
```suggestion
static char* nr_monolog_build_message(const size_t argc NRUNUSED, /* unused when $context is ignored */
                                      NR_EXECUTE_PROTO TSRMLS_DC) {
```

Reply:
I've never used NRUNUSED, but I think this should work?

Reply:
I was thinking about `NRUNUSED` attribute in argument list option but IMHO 
```
#if !HAVE_CONTEXT_IN_MESSAGE
  /* Make the compiler happy - argc is not used when $context is ignored */
  (void)argc;
#endif
```
is more explicit than having `NRUNUSED` attribute on an argument that can be used.

Reply:
I'm okay with either. NRUNUSED is cleaner but the preprocessor is more explicit.